### PR TITLE
Add locale:import:all - imports multiple custom translation po files

### PR DESCRIPTION
**Pull request by andypost**
Supersedes #4251

### CODE REVIEW

File: src/Commands/core/LocaleCommands.php
Comment (Sutharsan) [inline src/Commands/core/LocaleCommands.php]: The word 'custom' is not applicable here. 
I suggest to add that the directory is _not_ searched recursively.
I also suggest to add that the translation language is automatically determined based on the file name (some-name.langcode.po).

File: src/Commands/core/LocaleCommands.php
Comment (Sutharsan) [inline src/Commands/core/LocaleCommands.php]: No directory is provided in the example. /var/www/translations/custom

File: src/Commands/core/LocaleCommands.php
Comment (Sutharsan) [inline src/Commands/core/LocaleCommands.php]: 'destination' is not applicable here.

File: src/Commands/core/LocaleCommands.php
Comment (Sutharsan) [inline src/Commands/core/LocaleCommands.php]: 'destination' is not applicable here.

File: src/Commands/core/LocaleCommands.php
Comment (Sutharsan) [inline src/Commands/core/LocaleCommands.php]: Add an explanation of the required file name pattern. `some-name.langcode.po`

**Review by Sutharsan — CHANGES_REQUESTED** [review]
Initial review of the code. I did not test the command yet.

<strike>Note: Performing the same import with locale:import is possible but requires bash scripting.</strike>

File: src/Commands/core/LocaleCommands.php
Comment (Sergey-Orlov) [reply src/Commands/core/LocaleCommands.php]: updated description for this example

File: src/Commands/core/LocaleCommands.php
Comment (Sergey-Orlov) [reply src/Commands/core/LocaleCommands.php]: fixed

File: src/Commands/core/LocaleCommands.php
Comment (Sergey-Orlov) [reply src/Commands/core/LocaleCommands.php]: updated

File: src/Commands/core/LocaleCommands.php
Comment (Sergey-Orlov) [reply src/Commands/core/LocaleCommands.php]: updated

File: src/Commands/core/LocaleCommands.php
Comment (Sergey-Orlov) [reply src/Commands/core/LocaleCommands.php]: Added details about pattern

**Comment by weitzman** [comment]
@Sutharsan Is all feedback addressed?

File: src/Commands/core/LocaleCommands.php
Comment (Sutharsan) [inline src/Commands/core/LocaleCommands.php]: The other if conditions in this foreach loop use `continue` to skip processing the current `$file`. I agree with using this code style (early return). I'd like to use it consistently and apply it to this condition too, and remove these nested if conditions. With this being done, the comment 'Ensure we have the file ...' can go away. It is pretty obvious what is going on here.

**Comment by Sutharsan** [comment]
I have tested the code and all works as expected. 

@weitzman If you provide an array of command aliases, drush shows all aliases except the first. Is this intended behavior?  (I see the same behavior with existing commands)
Since this is a new command why should we provide aliases? No BC required, no short code.

The command code:  
```
#[CLI\Command(name: self::IMPORT_ALL, aliases: ['locale-import-all', 'locale:import:all'])]
```

The output of `drush list`:
```
locale:                                                                                                                        
  locale:check                                 Checks for available translation updates.                                       
  locale:clear-status                          Clears the translation status.                                                  
  locale:export                                Exports to a gettext translation file.                                          
  locale:import                                Imports to a gettext translation file.                                          
  locale:import-all (locale:import:all)        Imports multiple translation files from the defined directory.                  
  locale:update                                Imports the available translation updates.                                      
```

File: src/Commands/core/LocaleCommands.php
Comment (weitzman) [inline src/Commands/core/LocaleCommands.php]:
```suggestion
    public function importAll($directory, $options = ['type' => self::REQ, 'override' => self::REQ])
```
The meaning of REQ is that a value must be supplied when the option is used.

File: src/Commands/core/LocaleCommands.php
Comment (weitzman) [inline src/Commands/core/LocaleCommands.php]: Use suggestionValues param as per https://github.com/drush-ops/drush/blob/166af7acb72ad256cbc8c33fba58f5981683cd5a/src/Commands/config/ConfigCommands.php#L146
https://github.com/weitzman/drush/blob/166af7acb72ad256cbc8c33fba58f5981683cd5a/src/Commands/config/ConfigCommands.php#L146. Helps with shell completion.

**Review by weitzman — COMMENTED** [review]
Yeah, we should omit "dash" aliases for new commands.

**Comment by weitzman** [comment]
> If you provide an array of command aliases, drush shows all aliases except the first. Is this intended behavior? 

Looks like a bug. I think we should show only the first alias. Otherwise the first column gets too wide.

**Comment by weitzman** [comment]
[Actually ListCommands hides 'dash' aliases](https://github.com/drush-ops/drush/blob/c2b51cf5c0b537cc989513ccaa05e3e229104d4f/src/Commands/help/ListCommands.php#L132-L139).

File: src/Commands/core/LocaleCommands.php
Comment (Sergey-Orlov) [reply src/Commands/core/LocaleCommands.php]: fixed

File: src/Commands/core/LocaleCommands.php
Comment (Sergey-Orlov) [reply src/Commands/core/LocaleCommands.php]: done

File: src/Commands/core/LocaleCommands.php
Comment (Sergey-Orlov) [reply src/Commands/core/LocaleCommands.php]: done, for **override** option as well

**Review by weitzman — APPROVED** [review]
LGTM. @Sutharsan will merge when satisfied.

**Comment by Sergey-Orlov** [comment]
rebased forked branch, so all checks should be fine now

File: src/Commands/core/LocaleCommands.php
Comment (Sutharsan) [inline src/Commands/core/LocaleCommands.php]:
```suggestion
                $this->logger->warning(dt('Can not read file @file.', ['@file' => $file]));
```

File: src/Commands/core/LocaleCommands.php
Comment (Sutharsan) [inline src/Commands/core/LocaleCommands.php]:
```suggestion
                $this->logger->warning(dt('Language @language does not exist for file @file', [
```

**Review by Sutharsan — CHANGES_REQUESTED** [review]
I found two sentences that need some improvement.

**Comment by Sutharsan** [comment]
Once this is fixed, I will merge the PR.

**Comment by Sergey-Orlov** [comment]
> Once this is fixed, I will merge the PR.

Fixed and rebased again

**Review by Sutharsan — APPROVED** [review]

**Comment by Sutharsan** [comment]
Thanks all for working on this.

**Pull request merged**
